vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -112,12 +112,10 @@ def _create_scale_param(
                                       dtype=torch.float32),
                           requires_grad=False)
         layer.register_parameter(scale_name, scale)
-        set_weight_attrs(
-            scale, {
-                **extra_weight_attrs,
-                "fp8_scales_shard_indexer":
-                self.scales_shard_indexer,
-            })
+        set_weight_attrs(scale, {
+            **extra_weight_attrs,
+            "is_per_tensor_scale": True,
```

Comment:
I feel per-tensor scale is more commonly used than per-channel. Do you think it'd be more concise if we use "is_per_channel_scale" and only set the value for w8a8?

Reply:
* The `per-tensor-scase` case is the special case for weight loading, because we need to handle loading 1 scale into an array of N scales since we have fusion for QKV and MLP
* We actually don't have a problem for channelwise, since we just make a buffer that is `sum(partition_sizes)` and load with the same logic we use for weights
* So we need an attribute for `is_per_tensor_scale` that the weight loader can branch off to do the indexing properly

Reply:
Basically, we used to branch on `fp8_weight_shard_indexer` or something like that ... this is the replacement + the functionality is shared across both int8 and fp8 for this case since its the same logic

Reply:
That's a fair point. I guess I was just a bit confused about the naming is_per_tensor_scale being used as a special case for fused layers.

Reply:
It’s hard to pick a name. I’ll think of something else